analyzers/src/SonarAnalyzer.CSharp/Rules/EqualityOnFloatingPoint.cs

```diff
@@ -64,80 +64,71 @@ private static void CheckLogicalExpression(SyntaxNodeAnalysisContext context)
         }
 
-        private static BinaryExpressionSyntax TryGetBinaryExpression(ExpressionSyntax expression)
-        {
-            return expression.RemoveParentheses() as BinaryExpressionSyntax;
-        }
-
-        private static bool IsIndirectInequality(BinaryExpressionSyntax binaryExpression, BinaryExpressionSyntax right,
-            BinaryExpressionSyntax left, SyntaxNodeAnalysisContext context)
-        {
-            return binaryExpression.IsKind(SyntaxKind.LogicalOrExpression) &&
-                   HasAppropriateOperatorsForInequality(right, left) &&
-                   HasFloatingType(right.Right, right.Left, context.SemanticModel);
-        }
-
-        private static bool IsIndirectEquality(BinaryExpressionSyntax binaryExpression, BinaryExpressionSyntax right,
-            BinaryExpressionSyntax left, SyntaxNodeAnalysisContext context)
-        {
-            return binaryExpression.IsKind(SyntaxKind.LogicalAndExpression) &&
-                   HasAppropriateOperatorsForEquality(right, left) &&
-                   HasFloatingType(right.Right, right.Left, context.SemanticModel);
-        }
-
-        private static bool HasFloatingType(ExpressionSyntax right, ExpressionSyntax left, SemanticModel semanticModel)
-        {
-            return IsExpressionFloatingType(right, semanticModel) ||
-                IsExpressionFloatingType(left, semanticModel);
-        }
-
-        private static bool IsExpressionFloatingType(ExpressionSyntax expression, SemanticModel semanticModel)
-        {
-            return semanticModel.GetTypeInfo(expression).Type.IsAny(KnownType.FloatingPointNumbers);
-        }
-
-        private static bool HasAppropriateOperatorsForEquality(BinaryExpressionSyntax right, BinaryExpressionSyntax left)
-        {
-            return new[] {right.OperatorToken.Kind(), left.OperatorToken.Kind()}
-                .Intersect(new[] {SyntaxKind.LessThanEqualsToken, SyntaxKind.GreaterThanEqualsToken})
-                .Count() == 2;
-        }
-        private static bool HasAppropriateOperatorsForInequality(BinaryExpressionSyntax right, BinaryExpressionSyntax left)
-        {
-            return new[] { right.OperatorToken.Kind(), left.OperatorToken.Kind() }
-                .Intersect(new[] { SyntaxKind.LessThanToken, SyntaxKind.GreaterThanToken })
-                .Count() == 2;
-        }
+        // All floating point types that suffer from equivalence problems. These are all .net floating point types except decimal.
+        // Reason: The Equals documentation for decimal does not have a "Precision in Comparisons" section as the other floating point types.
+        // Power 2 based types like double implement IFloatingPointIeee754 but power 10 based decimal not (implements IFloatingPoint).
+        // Ieee754 also allows power 10 based representations but uses another representation than the .Net decimal type.
+        // IFloatingPointIeee754 defines Epsilon which indicates problems with equivalence checking.
```

Comment:
I suggest some simplifications/improvements.

I find the `Ieee754` mention confusing, and I'd rather avoid mentioning it.

```suggestion
        // Returns true for the floating point types that suffer from equivalence problems. All .NET floating point types have this problem except `decimal.`
        // -  the documentation for the `decimal.Equals()` method does not have a "Precision in Comparisons" section as the other .NET floating point types.
        // - Power-2-based types like `double` implement `IFloatingPointIeee754`, but power-10-based `decimal` implements `IFloatingPoint`.
        // - `IFloatingPointIeee754` defines `Epsilon` which indicates problems with equivalence checking.
```